addons/mail/static/src/utils/common/hooks.js

```diff
@@ -442,15 +443,28 @@ export function useSelection({ refName, model, preserveOnClickAwayPredicate = ()
 }
 
 export function useMessageEdition() {
+    /** @type {import("@mail/core/common/store_service").Store} */
+    const store = useService("mail.store");
     const state = useState({
         /** @type {import('@mail/core/common/composer').Composer} */
         composerOfThread: null,
         /** @type {import('@mail/core/common/message_model').Message} */
         editingMessage: null,
-        enterEditMode(message) {
+        async enterEditMode(message) {
             const rawMsg = toRaw(message);
+            const doc = createDocumentFragmentFromContent(rawMsg.body);
+            const validChannels = (
```

Comment:
Technically adding `async` on function doesn't follow stable policy:
- non-official overrides probably don't have `await` thus any override that expects to do something after super-call will break from adding this `async` (since calling an async function adds a micro-tick in its inner-code!)

I don't think this is critical, but if we want to play safe, it'd be best to make another method that is async and call this one. Unfortunately we'd have the right async function with different name than what we have in 19.0 / master, but that's the challenge with following stable policy.